manager/integration/tests/test_ha.py

```diff
@@ -2083,31 +2082,30 @@ def test_auto_remount_with_subpath(
             client, core_api, statefulset, sts_name, storage_class,
             data_path=data_path)
 
-    vol = client.by_id_volume(vol_name)
-    vol.updateDataLocality(dataLocality="best-effort")
-    wait_for_volume_healthy(client, vol_name)
-    common.wait_for_pod_remount(core_api, pod_name, chk_path=data_path)
-
     crash_count = 3
     for _ in range(crash_count):
         vol = client.by_id_volume(vol_name)
         rim_name = vol.replicas[0].instanceManagerName
         delete_and_wait_pod(core_api, rim_name,
                             namespace='longhorn-system',
                             wait=False)
+        time.sleep(2)  # Wait a bit for the deletion to make impact on volume
         wait_for_volume_healthy(client, vol_name)
         common.wait_for_pod_remount(core_api, pod_name, chk_path=data_path)
         expect_md5sum = get_pod_data_md5sum(core_api, pod_name, data_path)
         assert expect_md5sum == md5sum
 
     vol = client.by_id_volume(vol_name)
     vol.updateReplicaCount(replicaCount=3)
-    wait_for_rebuild_complete(client, vol_name)
+    wait_for_volume_replica_count(client, vol_name, 3)
```

Comment:
yes. The intention is that we want to have 3 healthy replicas before continue to next step and kill the engine. If we call `wait_for_rebuild_complete()`, it might continue after finish rebuilding the 1st replicas (I saw this).

This doesn't impact the correctness of the test but it is indeterministic.

Reply:
Does `wait_for_volume_replica_count` guarantee the replicas are healthy?

Reply:
[wait_for_volume_replica_count](https://github.com/longhorn/longhorn-tests/blob/8c55e55b8f4de43c1ffd8267da332be9ef31e9a2/manager/integration/tests/common.py#L1460) doesn't guarantee the replicas are healthy

Reply:
Do we need the replicas to be healthy to continue with the steps below?

Reply:
You are right, we don't need to. We don't even need to increase number of replicas. It doesn't impact the correctness of the test.  But if the test's intention was to have a volume with 3 healthy replicas and then kill the engine, we need to wait for the replicas to become healthy